patchwork-events-world/src/main/java/com/patchworkmc/mixin/event/world/MixinServerWorld.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2020, 2019-2020
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package com.patchworkmc.mixin.event.world;
+
+import java.util.function.BiFunction;
+
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.event.world.WorldEvent;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.util.profiler.Profiler;
+import net.minecraft.world.chunk.ChunkManager;
+import net.minecraft.world.dimension.Dimension;
+import net.minecraft.world.dimension.DimensionType;
+import net.minecraft.world.level.LevelInfo;
+import net.minecraft.world.level.LevelProperties;
+import net.minecraft.world.World;
+import net.minecraft.server.world.ServerWorld;
+
+import com.patchworkmc.impl.event.world.WorldEvents;
+
+@Mixin(ServerWorld.class)
+public abstract class MixinServerWorld extends World {
+	protected MixinServerWorld(LevelProperties levelProperties, DimensionType dimensionType, BiFunction<World, Dimension, ChunkManager> chunkManagerProvider, Profiler profiler, boolean isClient) {
+		super(levelProperties, dimensionType, chunkManagerProvider, profiler, isClient);
+	}
+
+	@Inject(method = "save", at = @At(value = "INVOKE", target = "net/minecraft/server/world/ServerChunkManager.save (Z)V"))
+	private void hookSave(CallbackInfo info) {
+		ServerWorld world = (ServerWorld) (Object) this;
+		MinecraftForge.EVENT_BUS.post(new WorldEvent.Save(world));
+	}
+
+	// TODO: consider adding a shift to before obtaining the ChunkManager to match forge more closely
+	// I don't think it'll make much of a difference.
```

Comment:
Technically this should be a `FIELD` injection before the `GETFIELD` instruction.

Reply:
Right, really the only reason I didn't do that is because chunkManager is accessed 3 times in the function, while getBiomeSource is only called once. We could setup a slice or an ordinal for that, but if this does the job...